src/sentry/tasks/post_process.py

```diff
@@ -595,6 +595,37 @@ def process_rules(job: PostProcessJob) -> None:
     return
 
 
+def process_code_mappings(job: PostProcessJob) -> None:
+    if job["is_reprocessed"]:
+        return
+
+    from sentry.tasks.derive_code_mappings import derive_code_mappings
+
+    try:
+        event = job["event"]
+        project = event.project
+
+        cache_key = f"code-mappings:{project.id}"
+        project_queued = cache.get(cache_key)
+        if project_queued is None:
+            cache.set(cache_key, True, 3600)
```

Comment:
If we use the project_id as the controlling mechanism here, we will need to control fetching the trees per org on the other task at the org id level. Otherwise, two events from two projects will trigger `get_trees_for_org` twice in the same hour. Now that I see this here I can look into adding it tomorrow morning or feel free to add it. Let's just coordinate it about it.

Reply:
It _may_ make sense to add logging in case we want to debug the system:
`logger.info(f"derive_code_mappings: Events from {project.id} will not have code derivation until {date_time_here}"`

Reply:
Added logging, good point!

Let's talk about your other point tomorrow. Our worst case scenario here is an org with a large number of projects (N) that all get events each hour.  I'm not sure how to prevent `get_trees_for_org` being called N times.

Reply:
What could happen within an hour for events for the same org:
* Event A1 for project A
  * We try to derive code and `get_trees_for_org` gets called 
* Event A2 for project A (cache prevents calling task)
* Event B1 for project B
  *  We try to derive code and `get_trees_for_org` gets called a 2nd time

We could check the cache key here (or an extra one) to only allow one event processed per hour per org for now so we can go live.

I will work today on adding a way to track what is the current GH api limit and how to control when `get_trees_for_org` gets called. I need to research on memcache and the potential for OOM since the trees for org object becomes quite large.